src/sentry/eventstream/kafka/backend.py

```diff
@@ -148,6 +148,18 @@ def _send(
             # flush() is a convenience method that calls poll() until len() is zero
             self.producer.flush()
 
+    @staticmethod
+    def _key(headers: Optional[Mapping[str, str]], project_id: str) -> Optional[str]:
+        if "transaction_forwarder" in headers:
+            message_type = "transaction" if headers["transaction_forwarder"] == "0" else "error"
+            if killswitch_matches_context(
+                "kafka.send-project-events-to-random-partitions",
+                {"project_id": project_id, "message_type": message_type},
+            ):
+                return None
```

Comment:
Why was this implemented as a killswitch instead of a standard option ?

Reply:
They function the same but killswitches give us metrics so we have some insight into how the options are running.

Reply:
The kill switch logic gives us metrics so we can see how often the flag is matching vs. not matching, and how many times it's considered automatically